types/index.d.ts

```diff
@@ -452,6 +461,9 @@ declare module "miragejs/server" {
       handler?: RouteHandler<Registry, Response>,
       options?: HandlerOptions
     ): void;
+    
+    /** Define multiple shorthands for a given resource */
+    resource(model: string, options?: ResourceOptions): void;
```

Comment:
Let's restrict the modelName to valid values, like what's done in `createList` above.  

```suggestion
    resource<K extends keyof Registry>(modelName: K, options?: ResourceOptions): void;
```